snakemake/io.py

```diff
@@ -835,7 +847,16 @@ def get_missing():
 
 
 def get_wildcard_names(pattern):
-    return set(match.group("name") for match in _wildcard_regex.finditer(pattern))
+    match = set(match.group("name") for match in _wildcard_regex.finditer(pattern))
+    illegal_name = False
+    if not match:
+        match = set(
+            match.group("name")
+            for match in _illegal_wildcard_name_regex.finditer(pattern)
+        )
+        if match:
+            illegal_name = True
+    return (match, illegal_name)
```

Comment:
I think instead of having the logic here, the warning should be done inside of `_IOFile.check()`.

Reply:
Hmmm in my test case the workflow would fail before `_IOFile.check()` is called on the file path containing wildcards since the rule that depends on it won't be able to recognize it as a potential producer. Any ideas?